tests/flakes/flake-in-submodule.sh

```diff
@@ -0,0 +1,54 @@
+source common.sh
+
+# Tests that:
+# - flake.nix may reside inside of a git submodule
+# - the flake can access content outside of the submodule
+#
+#   rootRepo
+#   ├── root.nix
+#   └── submodule
+#       ├── flake.nix
+#       └── sub.nix
+
+
+requireGit
+
+clearStore
+
+# Submodules can't be fetched locally by default.
+# See fetchGitSubmodules.sh
+export XDG_CONFIG_HOME=$TEST_HOME/.config
+git config --global protocol.file.allow always
+
+
+rootRepo=$TEST_ROOT/rootRepo
+subRepo=$TEST_ROOT/submodule
+
+rm -rf $rootRepo $subRepo $TEST_HOME/.cache/nix
+
```

Comment:
```suggestion
```

The test root starts clean, containing just a `test-home`.
I can appreciate defensive coding, but this is a bit too much imo.